pkg/destroy/gcp/cloudcontroller.go

```diff
@@ -0,0 +1,201 @@
+// https://github.com/openshift/kubernetes/blob/1e5983903742f64bca36a464582178c940353e9a/pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go#L289
+func (o *ClusterUninstaller) discoverLoadBalancerResources(loadBalancerName string) error {
+	loadBalancerNameFilter := fmt.Sprintf("name eq \"%s\"", loadBalancerName)
+
+	// Discover associated addresses: loadBalancerName
+	found, err := o.listAddressesWithFilter("items(name),nextPageToken", loadBalancerNameFilter, nil)
+	if err != nil {
+		return err
+	}
+	o.insertPendingItems("address", found)
+
+	// Discover associated firewall rules: loadBalancerName
+	found, err = o.listFirewallsWithFilter("items(name),nextPageToken", loadBalancerNameFilter, nil)
+	if err != nil {
+		return err
+	}
+	o.insertPendingItems("firewall", found)
+
+	// Discover associated firewall rules: loadBalancerName-hc
+	filter := fmt.Sprintf("name eq \"%s-hc\"", loadBalancerName)
+	found, err = o.listFirewallsWithFilter("items(name),nextPageToken", filter, nil)
+	if err != nil {
+		return err
+	}
+	o.insertPendingItems("firewall", found)
+
+	// Discover associated firewall rules: k8s-fw-loadBalancerName
+	filter = fmt.Sprintf("name eq \"k8s-fw-%s\"", loadBalancerName)
+	found, err = o.listFirewallsWithFilter("items(name),nextPageToken", filter, nil)
+	if err != nil {
+		return err
+	}
+	o.insertPendingItems("firewall", found)
+
+	// Discover associated firewall rules: k8s-loadBalancerName-http-hc
+	filter = fmt.Sprintf("name eq \"k8s-%s-http-hc\"", loadBalancerName)
+	found, err = o.listFirewallsWithFilter("items(name),nextPageToken", filter, nil)
+	if err != nil {
+		return err
+	}
+	o.insertPendingItems("firewall", found)
+
+	// Discover associated forwarding rules: loadBalancerName
+	found, err = o.listForwardingRulesWithFilter("items(name),nextPageToken", loadBalancerNameFilter, nil)
+	if err != nil {
+		return err
+	}
+	o.insertPendingItems("forwardingrule", found)
+
+	// Discover associated health checks: loadBalancerName
+	found, err = o.listHealthChecksWithFilter("items(name),nextPageToken", loadBalancerNameFilter, nil)
+	if err != nil {
+		return err
+	}
+
+	o.insertPendingItems("healthcheck", found)
+
+	// Discover associated http health checks: loadBalancerName
+	found, err = o.listHTTPHealthChecksWithFilter("items(name),nextPageToken", loadBalancerNameFilter, nil)
+	if err != nil {
+		return err
+	}
+	o.insertPendingItems("httphealthcheck", found)
+
+	return nil
+}
+
+// discoverCloudControllerResources finds resources associated with internal load balancers
+// created by the kube cloud controller. It first finds instance groups associated with instances
+// belonging to this cluster, then finds backend resources that point to these instance groups.
+// For each of those backend services, resources like forwarding rules, firewalls, health checks and
+// backend services are added to pendingItems
+func (o *ClusterUninstaller) discoverCloudControllerResources() error {
```

Comment:
this function is doing more than just internal load balancer...

also any failure to discover the internal LB tied resources, causes the external LB resources to not be discovered... this tie-in is not great.

Reply:
There is a lot of duplication between internal and external cloud controller resources... so it makes sense to keep this together. Would it be better to aggregate the errors instead of gating on them?

Reply:
Keep in mind... "not found" does not appear to be an err... so this only happens if there is another problem... in which case we might actually want to gate?